core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowResource.scala

```diff
@@ -443,4 +444,107 @@ class WorkflowResource {
       workflowDao.update(userWorkflow)
     }
   }
+
+  /**
+    * This method performs a full-text search in the content column of the
+    * workflow table for workflows that match the specified keywords.
+    *
+    * This method utilizes MySQL Boolean Full-Text Searches
+    * reference: https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html
+    * @param sessionUser The authenticated user.
+    * @param keywords    The search keywords.
+    * @return A list of workflows that match the search term.
+    */
+  @GET
+  @Path("/search")
+  def searchWorkflows(
+      @Auth sessionUser: SessionUser,
+      @QueryParam("query") keywords: java.util.List[String]
+  ): List[DashboardWorkflowEntry] = {
+    val user = sessionUser.getUser
+    if (keywords.size() == 0) {
+      return List.empty[DashboardWorkflowEntry]
+    }
+    // make sure keywords don't end with "+-()<>~*\"", these are reserved for SQL full-text boolean operator
+    val forbiddenChars: Set[Char] = "+-()<>~*\"".toSet
```

Comment:
There is a suggestion [here](https://stackoverflow.com/questions/26507087/escape-string-for-use-in-mysql-fulltext-search) that says special characters are not indexed and therefore should simply be replaced with spaces. 
Perhaps if the keyword is `jingchf@uci.edu`, you can put `jingchf` and `uci.edu` in the query and mysql will actually return the correct results because it does not treat the whole email address as a single word.

Reply:
I think replacing reserved characters with space is a good idea!